src/accountsdb/db.zig

```diff
@@ -358,27 +459,54 @@ pub const AccountsDB = struct {
             };
             // if this is hit, its likely an old snapshot
             if (file_infos.items.len != 1) {
-                std.debug.panic("incorrect file_info count for slot {d}, likley trying to load from an unsupported snapshot\n", .{slot});
+                self.logger.errf("incorrect file_info count for slot {d}, likley trying to load from an unsupported snapshot\n", .{slot});
+                return error.InvalidSnapshot;
             }
             const file_info = file_infos.items[0];
-            if (file_info.id != accounts_file_id) {
-                std.debug.panic("file_info.id ({d}) != accounts_file_id ({d})\n", .{ file_info.id, accounts_file_id });
+            if (file_info.id != file_id_usize) {
+                self.logger.errf("file_id from metadata for slot {d} doesnt match file_id from filename: {d} vs {d}\n", .{ slot, file_info.id, file_id_usize });
+                return error.InvalidSnapshotFormat;
             }
 
             // read accounts file
             const abs_path = try std.fmt.bufPrint(&buf, "{s}/{s}", .{ accounts_dir_path, file_name });
             const accounts_file_file = try std.fs.cwd().openFile(abs_path, .{ .mode = .read_write });
             var accounts_file = AccountFile.init(accounts_file_file, file_info, slot) catch |err| {
-                std.debug.panic("failed to *open* AccountsFile {s}: {s}\n", .{ file_name, @errorName(err) });
+                self.logger.errf("failed to *open* AccountsFile {s}: {s}\n", .{ file_name, @errorName(err) });
+                return err;
             };
 
-            // validate and count here for prealloc
-            self.account_index.validateAccountFile(&accounts_file, bin_counts, refs_ptr) catch |err| {
-                std.debug.panic("failed to *sanitize* AccountsFile: {d}.{d}: {s}\n", .{ accounts_file.slot, accounts_file.id, @errorName(err) });
+            const reference_slice_start = references.items.len;
+            self.account_index.validateAccountFile(&accounts_file, bin_counts, &references) catch |err| {
+                if (err == error.OutOfReferenceMemory) {
```

Comment:
8ed4ad4

Reply:
To be clear, my intended meaning was that it's safer to use unqualified error values as switch prongs, because PTR won't be performed - ie, either use compare using `==` against a qualified error value, or switch on unqualified error prongs.

That said, it's fine, won't block on this small detail.

Reply:
do you have an example of both scenarios mentioned in the 'ie'? - not 100% on what you mean

Reply:
```zig
const A = error { Foo };
fn a() A!void { return error.Foo; } // inference is fine, it's type-checked by the return type
// fn b() !void { return A.F00; } // caught by the compiler, there isn't a F00 in A
// fn c() !void {  }
fn d() !void { return error.F00; } // potentially silent killer if you were expecting A.Foo

test {
    if (A.Foo == error.F00) {} // this compiles, the resolved type used for comparison is `error{ Foo, F00 }`
    switch (A.Foo) {
        // error.F00 => {}, // this is a compile error, the type of the prong value must coerce to the operand; error.F00 is not in A, so it will not coerce
        error.Foo => {}, // this is fine unqualified, because it you were to make a typo, you'd get a compile error for the above reason
    }
}
```
Also note: `A.Foo == error.Foo`, because conceptually, errors are all part of one big enum, and error sets are just subsets of the overarching superset. So whether you qualify it only influences the type checking boundary, not the value itself.